src/firstboot/firstboot.c

```diff
@@ -672,6 +737,18 @@ static int process_root_password(void) {
         if (lock < 0)
                 return log_error_errno(lock, "Failed to take a lock: %m");
 
+        if (arg_delete_root_password) {
+                const char *etc_passwd = prefix_roota(arg_root, "/etc/passwd");
```

Comment:
hmm, please do that in two lines. prefix_roota() uses alloca() internally, and according to C specs mixing alloca() and other stuff that touches the stack in one expression is not portable at all... i.e. this allocates one stack variable afte rall, and then assigns memory also allocated from the stack to it...